providers/enchant_nuspell.cpp

```diff
@@ -0,0 +1,445 @@
+	 */
+	char * enchant_prefix = enchant_get_prefix_dir();
+	if (enchant_prefix) {
+		tmp = g_build_filename(enchant_prefix, "share", "enchant", "nuspell", nullptr);
+		dirs.push_back (tmp);
+		g_free(enchant_prefix);
+		g_free(tmp);
+	}
+
+	/* Hunspell paths are used, therefore is ENCHANT_NUSPELL_DICT_DIR
+	 * irrelevant. Hence, the following paths are not to be considered:
+	 * /usr/local/share/nuspell and /usr/share/nuspell
+	 */
+}
+
+static void
+s_buildHashNames (vector<string> & names, const char * dict)
+{
+	names.clear ();
+
+	vector<string> dirs;
+	s_buildDictionaryDirs (dirs);
+
+	char *dict_dic = g_strconcat(dict, ".dic", nullptr);
+	for (size_t i = 0; i < dirs.size(); i++) {
+		char *tmp = g_build_filename (dirs[i].c_str(), dict_dic, nullptr);
+		names.push_back (tmp);
+		g_free (tmp);
+	}
+
+	g_free(dict_dic);
+}
+
+static const string
+s_correspondingAffFile(const string & dicFile)
+{
+	string aff = dicFile;
+	aff.replace(aff.end()-3,aff.end(), "aff");
+	return aff;
+}
+
+static bool
+s_fileExists(const string & file)
+{
+	return g_file_test(file.c_str(), G_FILE_TEST_EXISTS) != 0;
+}
+
+static bool is_plausible_dict_for_tag(const char *dir_entry, const char *tag)
+{
+	const char *dic_suffix = ".dic";
+	size_t dic_suffix_len = strlen(dic_suffix);
+	size_t dir_entry_len = strlen(dir_entry);
+	size_t tag_len = strlen(tag);
+
+	if (dir_entry_len - dic_suffix_len < tag_len)
+		return false;
+	if (strcmp(dir_entry+dir_entry_len-dic_suffix_len, dic_suffix) != 0)
+		return false;
+	if (strncmp (dir_entry, tag, tag_len) != 0)
+		return false;
+	//e.g. requested dict for "fi",
+	//reject "fil_PH.dic"
+	//allow "fi-FOO.dic", "fi_FOO.dic", "fi.dic", etc.
+	if (!ispunct(dir_entry[tag_len]))
+		return false;
+	return true;
+}
+
+static char *
+nuspell_request_dictionary (const char * tag)
+{
+	vector<string> names;
+
+	s_buildHashNames (names, tag);
+
+	for (size_t i = 0; i < names.size (); i++) {
+		if (g_file_test(names[i].c_str(), G_FILE_TEST_EXISTS) &&
+		    s_fileExists(s_correspondingAffFile(names[i]))) {
+			return strdup (names[i].c_str());
+		}
+	}
+
+	vector<string> dirs;
+	s_buildDictionaryDirs (dirs);
+
+	for (size_t i = 0; i < dirs.size(); i++) {
+		GDir *dir = g_dir_open (dirs[i].c_str(), 0, nullptr);
+		if (dir) {
+			const char *dir_entry;
+			while ((dir_entry = g_dir_read_name (dir)) != NULL) {
+				if (is_plausible_dict_for_tag(dir_entry, tag)) {
+					char *dict = g_build_filename (dirs[i].c_str(),
+								       dir_entry, nullptr);
+					if(s_fileExists(s_correspondingAffFile(dict))) {
+						g_dir_close (dir);
+						return dict;
+					}
+					g_free(dict);
+				}
+			}
+
+			g_dir_close (dir);
+		}
+	}
+
+	return NULL;
+}
+
+bool
+NuspellChecker::requestDictionary(const char *szLang)
+{
+	char *dic = nuspell_request_dictionary (szLang);
+	if (!dic)
+		return false;
+	string aff(s_correspondingAffFile(dic));
+	if (!s_fileExists(aff))
+		return false;
+
+	auto path = string(dic);
+	path = path.substr(0, path.length()-4);
```

Comment:
Please don't just chop N characters off the string. If you require to match and remove a particular extension, or just any extension, then do that.

Reply:
Fixed in https://github.com/nuspell/enchant/commit/205c4b8423d3140a1c6f3e3e77f0841145c7fb69#diff-b4a4b7f548e6522299f21f4f5632ff1cR258